h2o-docs/src/product/performance-and-prediction.rst

```diff
@@ -631,10 +581,123 @@ This section provides examples of performing predictions in Python and R. Refer
+~~~~~~~~~~~~~~~~~~~~~
+
+In H2O-3, each returned H2OFrame has a specific shape (#rows, #features + 1). This includes a feature contribution column for each input feature, with the last column being the model bias (same value for each row). The sum of the feature contributions and the bias term is equal to the raw prediction of the model. Raw prediction of tree-based model is the sum of the predictions of the individual trees before the inverse link function is applied to get the actual prediction. For Gaussian distribution, the sum of the contributions is equal to the model prediction. 
+
+H2O-3 supports TreeSHAP for DRF, GBM, and XGBoost. For these problems, the ``predict_contributions`` returns a new H2OFrame with the predicted feature contributions - SHAP (SHapley Additive exPlanation) values on an H2O model.
+        
+**Note**: Multinomial classification models are currently not supported.
+
+Using the previous example, run the following to predict contributions:
+
+.. example-code::
+   .. code-block:: r
+  
+    # Predict the contributions using the GBM model and test data.
+    predict_contributions <- h2o.predict_contributions(model, prostate.test)
+    predict_contributions
+
+    AGE        RACE       PSA        GLEASON    BiasTerm
+    ---------  ---------- ---------  ---------  ----------
+    -0.3929753  0.02188157 0.3530045  0.5453218 -0.6589417
+    -0.6489378 -0.24417394 1.0434356  0.7937416 -0.6589417
+     0.3244801 -0.23901901 0.9877144  1.0463049 -0.6589417
+     0.9402978 -0.33412665 2.0499718  1.0571480 -0.6589417
+    -0.7762397  0.03393304 0.1952782  1.8620299 -0.6589417
+     0.5900557  0.03899451 0.6708371 -1.2606093 -0.6589417
+
+     [95 rows x 5 columns]
+
+
+   .. code-block:: python
+
+    # Predict the contributions using the GBM model and test data.
+    predict_contributions = model.predict_contributions(test)
+    predict_contributions
+
+    AGE          RACE        PSA        GLEASON    BiasTerm
+    -----------  ----------  ---------  ---------  ----------
+    -0.414587     0.0263119  -0.120703   0.407889   -0.581522
+     0.0913486    0.0250697  -0.746584   1.16642    -0.581522
+     0.565866     0.0603216   2.51301    0.739406   -0.581522
+    -0.670981     0.0210115   0.164873  -2.03487    -0.581522
+    -0.398603     0.0255295  -0.494069   0.537647   -0.581522
+     0.00915739   0.0458912   0.557667  -0.262171   -0.581522
+    -0.199497    -0.265438    2.18964    2.89974    -0.581522
+    -0.137073     0.0271401  -1.00939    1.47302    -0.581522
+     0.440857     0.0407717  -0.574498  -0.537758   -0.581522
+    -0.901466     0.0216657   0.453894  -2.39536    -0.581522
+
+    [58 rows x 5 columns]
+
+
+Predict Stage Probabilities
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use the ``staged_predict_proba`` function to predict class probabilities at each stage of an H2O Model. Note that this can only be used with GBM.
+
+Using the previous example, run the following to predict stage probabilities:
```

Comment:
Using the previous example, run the following to predict probabilities at each stage in the model.